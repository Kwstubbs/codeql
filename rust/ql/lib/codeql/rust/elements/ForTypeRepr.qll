// generated by codegen, do not edit
/**
 * This module provides the public class `ForTypeRepr`.
 */

private import internal.ForTypeReprImpl
import codeql.rust.elements.ForBinder
import codeql.rust.elements.TypeRepr

/**
 * A type with a higher-ranked `for` modifier. This is currently not valid Rust syntax (`for<...>` can
 * only be applied to traits to form a `TypeBound`).
 *
 * For example:
 * ```rust
 * fn foo(value: for<'a> usize) {}  // DOESN'T COMPILE
 * //            ^^^^^^^^^^^^^
 * ```
 */
final class ForTypeRepr = Impl::ForTypeRepr;
