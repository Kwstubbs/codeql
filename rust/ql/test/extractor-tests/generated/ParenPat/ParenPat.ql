// generated by codegen, do not edit
import codeql.rust.elements
import TestUtils

query predicate instances(ParenPat x, string hasPat__label, string hasPat) {
  toBeTested(x) and
  not x.isUnknown() and
  hasPat__label = "hasPat:" and
  if x.hasPat() then hasPat = "yes" else hasPat = "no"
}

query predicate getPat(ParenPat x, Pat getPat) {
  toBeTested(x) and not x.isUnknown() and getPat = x.getPat()
}
