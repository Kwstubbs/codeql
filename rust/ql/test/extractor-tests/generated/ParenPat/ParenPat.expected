instances
| gen_paren_pat.rs:7:9:7:11 | (...) | hasPat: | yes |
getPat
| gen_paren_pat.rs:7:9:7:11 | (...) | gen_paren_pat.rs:7:10:7:10 | x |
